api/core/agent/fc_agent_runner.py

```diff
@@ -21,6 +23,8 @@
 from core.tools.tool_engine import ToolEngine
 from models.model import Message
 
+posthog = Posthog(os.environ.get('POSTHOG_PROJECT_KEY'), host="https://us.i.posthog.com")
```

Comment:
you could initialize this once in the utils folder inside api/core and import it